model/distro/distro.go

```diff
@@ -388,8 +388,12 @@ func (d *Distro) ExecutableSubPath() string {
 	return filepath.Join(arch, d.BinaryName())
 }
 
-// HomeDir gets the absolute path to the home directory for this distro's user.
+// HomeDir gets the absolute path to the home directory for this distro's user
+// for non-legacy provisioned hosts.
 func (d *Distro) HomeDir() string {
+	if d.LegacyBootstrap() {
```

Comment:
why not invert these cases?

Reply:
I don't get what you mean. Legacy SSH is the exceptional edge case that has to be written exactly like this and depends on being run in a shell - all other provisioning methods can't rely on running in a shell, and therefore actually specify a path.

Anyways, it doesn't really matter because I'm going to remove this piece of code soon anyways.